coredocument/coredocument.go

```diff
@@ -149,6 +160,49 @@ func GetDocumentSigningTree(document *coredocumentpb.CoreDocument) (tree *proofs
 	return tree, nil
 }
 
+// GetDocumentSigningTree returns the merkle tree for the signing root
+func GetDocumentSigningTree(document *coredocumentpb.CoreDocument, dataRootCalculator func() ([]byte, error)) (tree *proofs.DocumentTree, err error) {
```

Comment:
why pass the function when you can actually pass the result i.e data root instead?

Reply:
A better way would be passing both coredoc root and data root to this function and returning a tree instead